inc/hclib_cpp.h

```diff
@@ -31,6 +31,38 @@ int current_worker();
 int num_workers();
 int get_num_places(place_type_t type);
 void get_places(place_t **pls, place_type_t type);
+place_t *get_current_place();
+place_t **get_children_of_place(place_t *pl, int *num_children);
+place_t *get_root_place();
+place_t **get_nvgpu_places(int *n_nvgpu_places);
+char *get_place_name(place_t *pl);
+
+#ifdef HC_CUDA
+template<typename T>
+T *allocate_at(place_t *pl, size_t nitems, int flags) {
```

Comment:
I think this came up in the last review too. If we're standardizing on hclib, we want to give consistently named functions that make it clear which package they belong to. For example, when you write an MPI program you don't call Send, Recv, Barrier, etc. You call MPI_Send, MPI_Receive, etc. The function names are always prefixed with the package name to prevent conflicts with other libraries. We're doing the same thing with the C API, so we'll always have hclib_allocate_at instead of just allocate_at. The C++ API is already exactly as you said, but in the namespace hclib.

If you were talking about the capitalization, those things were capitalized in the old HC because they were usually preprocessor macros, and that's the convention for macros. It wasn't a great idea, but I'm sure there was a good reason at the time. We shouldn't do that here, these are functions, typing in all caps is a pain, and I don't think there's a single standard API out there that uses all capitalized letters in their naming conventions.